### DISCUSSION TITLE
i've got a problem in setup

### DISCUSSION THREAD

**Discussion by erdus** [blockscout/blockscout#7174] [Support] [answered]
Hello. I can't install. Below is the error I got.

blockscout                 | Crash dump is being written to: erl_crash.dump...done
blockscout exited with code 1
blockscout                 | ERROR! Config provider Config.Reader failed with:
blockscout                 | ** (Code.LoadError) could not load /app/releases/5.1.1/config_helper.exs. Reason: enoent
blockscout                 |     (elixir 1.14.0) lib/code.ex:1807: Code.find_file!/2
blockscout                 |     (elixir 1.14.0) lib/code.ex:1184: Code.eval_file/2
blockscout                 |     /app/releases/5.1.1/runtime.exs:5: (file)
blockscout                 |     (elixir 1.14.0) src/elixir.erl:298: anonymous fn/4 in :elixir.eval_external_handler/1
blockscout                 |     (stdlib 4.0.1) erl_eval.erl:748: :erl_eval.do_apply/7
blockscout                 |     (stdlib 4.0.1) erl_eval.erl:136: :erl_eval.exprs/6
blockscout                 |     (elixir 1.14.0) src/elixir.erl:288: :elixir.eval_forms/3
blockscout                 |     (elixir 1.14.0) lib/module/parallel_checker.ex:100: Module.ParallelChecker.verify/1
blockscout                 |
blockscout                 | {"init terminating in do_boot",{#{'__exception__'=>true,'__struct__'=>'Elixir.Code.LoadError',file=><<"/app/releases/5.1.1/config_helper.exs">>,message=><<"could not load /app/releases/5.1.1/config_helper.exs. Reason: enoent">>,reason=>enoent},[{'Elixir.Code','find_file!',2,[{file,"lib/code.ex"},{line,1807}]},{'Elixir.Code',eval_file,2,[{file,"lib/code.ex"},{line,1184}]},{elixir_eval,'__FILE__',1,[{file,"/app/releases/5.1.1/runtime.exs"},{line,5}]},{elixir,'-eval_external_handler/1-fun-2-',4,[{file,"src/elixir.erl"},{line,298}]},{erl_eval,do_apply,7,[{file,"erl_eval.erl"},{line,748}]},{erl_eval,exprs,6,[{file,"erl_eval.erl"},{line,136}]},{elixir,eval_forms,3,[{file,"src/elixir.erl"},{line,288}]},{'Elixir.Module.ParallelChecker',verify,1,[{file,"lib/module/parallel_checker.ex"},{line,100}]}]}}
blockscout                 | init terminating in do_boot ({,[{Elixir.Code,find_file!,2,[{_},{_}]},{Elixir.Code,eval_file,2,[{_},{_}]},{elixir_eval,__FILE__,1,[{_},{_}]},{elixir,-eval_external_handler/1-fun-2-,4,[{_},{_}]},{erl_eval,do_apply,7,[{_},{_}]},{erl_eval,exprs,6,[{_},{_}]},{elixir,eval_forms,3,[{_},{_}]},{Elixir.Module.ParallelChecker,verify,1,[{_},{_}]}]})
blockscout                 |

**Comment by Mojmir-R** [comment]
Hey @erdus, thanks for reaching out, if there is no one able to advise here, you can also try the official Blockscout Discord: discord.gg/blockscout

**Comment by vbaranov** [comment]
@erdus Please try latest Blockscout release [5.1.2](https://github.com/blockscout/blockscout/releases/tag/v5.1.2-beta) or latest docker image tag. This issue fixed in both places.

**Reply by erdus** [reply]
I am sad. The error persists.
Where am I doing wrong?

I have done the following.
1- git clone https://github.com/blockscout/blockscout.git
2- cd blockscout/docker-compose
3- docker-compose up --build

I also downloaded version 5.1.2. I opened. I've tried. It didn't happen.

**Reply by erdus** [reply]
blockscout                 | {"init terminating in do_boot",{#{'__exception__'=>true,'__struct__'=>'Elixir.Code.LoadError',file=><<"/app/releases/5.1.2/config_helper.exs">>,message=><<"could not load /app/releases/5.1.2/config_helper.exs. Reason: enoent">>,reason=>enoent},[{'Elixir.Code','find_file!',2,[{file,"lib/code.ex"},{line,1807}]},{'Elixir.Code',eval_file,2,[{finit terminating in do_boot ({,[{Elixir.Code,find_file!,2,[{_},{_}]},{Elixir.Code,eval_file,2,[{_},{_}]},{elixir_eval,__FILE__,1,[{_},{_}]},{elixir,-eval_external_handler/1-fun-2-,4,[{_},{_}]},{erl_eval,do_apply,7,[{_},{_}]},{erl_eval,exprs,6,[{_},{_}]},{elixir,eval_forms,3,[{_},{_}]},{Elixir.Module.ParallelChecker,verify,1,[{_},{_}]}]})
blockscout                 | ile,"lib/code.ex"},{line,1184}]},{elixir_eval,'__FILE__',1,[{file,"/app/releases/5.1.2/runtime.exs"},{line,5}]},{elixir,'-eval_external_handler/1-fun-2-',4,[{file,"src/elixir.erl"},{line,298}]},{erl_eval,do_apply,7,[{file,"erl_eval.erl"},{line,748}]},{erl_eval,exprs,6,[{file,"erl_eval.erl"},{line,136}]},{elixir,eval_forms,3,[{file,"src/elixir.erl"},{line,288}]},{'Elixir.Module.ParallelChecker',verify,1,[{file,"lib/module/parallel_checker.ex"},{line,100}]}]}}
blockscout                 |
blockscout                 | Crash dump is being written to: erl_crash.dump...done

**Comment by vbaranov** [comment, marked as answer]
@erdus you're right building from Dockerfile was broken. It has been resolved with https://github.com/blockscout/blockscout/pull/7186. Could you please pull updates from master branch?

**Reply by erdus** [reply]
I cannot install. I'm about to lose my mind. Ubuntu 18 - 20 - 22 I tried them all. I tried manually, I tried with docker. I can't. Editing done. I can't do it again. Now I can't install the application that I installed in 10 minutes. It's not manual, it's not docker. What should I do?

`blockscout                 | Crash dump is being written to: erl_crash.dump...pplication_controller,"{application_start_failure,indexer,{{shutdown,{failed_to_start_child,'Elixir.Indexer.Supervisor',{shutdown,{failed_to_start_child,'Elixir.Indexer.Block.Catchup.Supervisor',{shutdown,{failed_to_start_child,'Elixir.Indexer.Block.Catchup.MissingRangesCollector',{{badmatch,{error,econnrefused}},[{'Elixir.Indexer.Block.Catchup.MissingRangesCollector',fetch_max_block_number,0,[{file,\"lib/indexer/block/catchup/missing_ranges_collector.ex\"},{line,197}]},{'Elixir.Indexer.Block.Catchup.MissingRangesCollector',get_initial_min_max,0,[{file,\"lib/indexer/block/catchup/missing_ranges_collector.ex\"},{line,114}]},{'Elixir.Indexer.Block.Catchup.MissingRangesCollector',default_init,0,[{file,\"lib/indexer/block/catchup/missing_ranges_collector.ex\"},{line,43}]},{'Elixir.Indexer.Block.Catchup.MissingRangesCollector',init,1,[{file,\"lib/indexer/block/catchup/missing_ranges_collector.ex\"},{line,25}]},{gen_server,init_it,2,[{file,\"gen_server.erl\"},{line,848}]},{gen_server,init_it,6,[{file,\"gen_server.erl\"},{line,811}]},{proc_lib,init_p_do_apply,3,[{file,\"proc_lib.erl\"},{line,240}]}]}}}}}}},{'Elixir.Indexer.Application',start,[normal,[]]}}}"}
blockscout                 | done
blockscout exited with code 1
`


`blockscout                 |     ** (EXIT) shutdown: failed to start child: Indexer.Block.Catchup.Supervisor
blockscout                 |         ** (EXIT) shutdown: failed to start child: Indexer.Block.Catchup.MissingRangesCollector
blockscout                 |             ** (EXIT) an exception was raised:
blockscout                 |                 ** (MatchError) no match of right hand side value: {:error, :econnrefused}
blockscout                 |                     (indexer 5.1.2) lib/indexer/block/catchup/missing_ranges_collector.ex:197: Indexer.Block.Catchup.MissingRangesCollector.fetch_max_block_number/0
blockscout                 |                     (indexer 5.1.2) lib/indexer/block/catchup/missing_ranges_collector.ex:114: Indexer.Block.Catchup.MissingRangesCollector.get_initial_min_max/0
blockscout                 |                     (indexer 5.1.2) lib/indexer/block/catchup/missing_ranges_collector.ex:43: Indexer.Block.Catchup.MissingRangesCollector.default_init/0
blockscout                 |                     (indexer 5.1.2) lib/indexer/block/catchup/missing_ranges_collector.ex:25: Indexer.Block.Catchup.MissingRangesCollector.init/1
blockscout                 |                     (stdlib 4.0.1) gen_server.erl:848: :gen_server.init_it/2
blockscout                 |                     (stdlib 4.0.1) gen_server.erl:811: :gen_server.init_it/6
blockscout                 |                     (stdlib 4.0.1) proc_lib.erl:240: :proc_lib.init_p_do_apply/3
postgres                   | 2023-04-01 12:11:48.489 UTC [2061] ERROR:  canceling statement due to user request
postgres                   | 2023-04-01 12:11:48.489 UTC [2061] STATEMENT:  SELECT t0."contract_address_hash" FROM "tokens" AS t0 WHERE ((t0."cataloged" = TRUE) AND (t0."updated_at" <= $1)) ORDER BY t0."updated_at"
postgres                   | 2023-04-01 12:11:48.496 UTC [2002] ERROR:  canceling statement due to user request
postgres                   | 2023-04-01 12:11:48.496 UTC [2002] STATEMENT:  SELECT DISTINCT ON (l0."block_number") l0."block_number" FROM "logs" AS l0 WHERE (l0."first_topic" = '0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef') AND (NOT (exists((SELECT st0."amount", st0."block_number", st0."log_index", st0."token_id", st0."amounts", st0."token_ids", st0."from_address_hash", st0."to_address_hash", st0."token_contract_address_hash", st0."transaction_hash", st0."block_hash", st0."inserted_at", st0."updated_at" FROM "token_transfers" AS st0 WHERE (st0."transaction_hash" = l0."transaction_hash") AND (st0."log_index" = l0."index")))))
`
